File: SRC/Service/Saras.eMarking/Saras.eMarking.Domain/ViewModels/Project/Setup/ProjectSchedule/ProjectScheduleModel.cs
using Newtonsoft.Json.Linq;
using Saras.eMarking.Domain.Extensions;
using Saras.eMarking.Domain.ViewModels.AuditTrail;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Saras.eMarking.Domain.ViewModels.Project.Setup.ProjectSchedule
{
    public class ProjectScheduleModel : IAuditTrail
    {
        [XssTextValidation]
        public string ScheduleName { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime ExpectedEndDate { get; set; }
        [Required]
        public TimeSpan StartTime { get; set; }
        [Required]
        public TimeSpan EndTime { get; set; } 
        public List<ProjectScheduleCalendarModel> ScheduleTimeModels { get; set; }
        public Dictionary<string, string> WorkingDaysConfig { get; set; }
        public JObject WorkingDaysConfigJson { get; set; }
        public System.DateTime? CalendarDate { get; set; }
        public byte DayType { get; set; }
        [Range(1, 2, ErrorMessage = "Please enter a value bigger than or equal to {1}")]
        public int? Confirmdialogeventvalue { get; set; }
        public bool IsUpdate { get; set; }
    }
}

File: SRC/Manifest/Saras.eMarkingClient/src/app/model/project/setup/project-schedule-model.ts
export class ProjectScheduleModel {
    StartDate!: any;
    ExpectedEndDate!: any;
    StartTime!: string;
    EndTime!: string;
    WorkingDaysConfig!: any;
    CalendarDate!: string;
    DayType!: number;
    SelectedDate!: string;
    Remarks!: string;
    WorkingDaysConfigJson!: JSON;
    scheduleTimeModels!: ProjectScheduleCalendarModel[];
    Confirmdialogeventvalue!: number;
    IsUpdate!: boolean;
}

export class ProjectScheduleCalendarModel {
    CalendarDate!: string;
    DayType!: number;
    StartTime!: string;
    EndTime!: string;
}

export class DayWiseScheduleModel {
    SelectedDate!: Date;
    StartTime!: string;
    EndTime!: string;
    DayType!: number;
    Remarks!: string;
    ProjectCalendarID!: number;
    StartDateTime!: Date;
    EndDateTime!: Date;
}